### PR TITLE
Improve public APIs to return only module specific errors

### PR DESCRIPTION
**Pull request by ayeshLK**
Part of [#1487](https://github.com/ballerina-platform/ballerina-standard-library/issues/1487)

## Checklist
- [X] Linked to an issue
- [X] Updated the changelog
- [X] Added tests

### CODE REVIEW

File: websubhub-ballerina/hub_client.bal [hunk 1]
```diff
@@ -61,10 +61,10 @@ public client class HubClient {
     # ```
     #
     # + msg - Content to be distributed to the topic subscriber 
-    # + return - An `error` if an exception occurred, a `websubhub:SubscriptionDeletedError` if the subscriber responded with `HTTP 410`,
+    # + return - An `Error` if an exception occurred, a `websubhub:SubscriptionDeletedError` if the subscriber responded with `HTTP 410`,
```
Comment (ldclakmal) [inline websubhub-ballerina/hub_client.bal:64]:
```suggestion
    # + return - An `websubhub:Error` if an exception occurred, a `websubhub:SubscriptionDeletedError` if the subscriber responded with `HTTP 410`,
```

File: websubhub-ballerina/utils.bal [hunk 2]
```diff
@@ -202,6 +202,15 @@ isolated function retrieveResponseBodyForFormUrlEncodedMessage(string payload) r
     return responsePayload;
 }
 
+isolated function retrieveHttpClient(string url, http:ClientConfiguration config) returns http:Client|Error {
+    http:Client|error clientEp = new (url, config);
+    if (clientEp is http:Client) {
+        return clientEp;
+    } else {
+        return error Error("Client initialization failed", clientEp);
+    }
```
Comment (ldclakmal) [inline websubhub-ballerina/utils.bal:211]:
```suggestion
    }
    return error Error("Client initialization failed", clientEp);
```

File: websubhub-ballerina/hub_client.bal [hunk 3]
```diff
@@ -157,37 +133,33 @@ isolated function retrieveContentType(string? contentType, string|xml|json|byte[
...
-# + payload - content-distribution request body
-# + return - `byte[]` containing the content signature or an `error` if there is any exception in the 
-#            function execution
-isolated function retrievePayloadSignature(string 'key, string|xml|json|byte[] payload) returns byte[]|error {
+isolated function retrievePayloadSignature(string contentType, string secret, string queryString, string|xml|json|byte[] payload) returns byte[]|error {
+    if contentType == mime:APPLICATION_FORM_URLENCODED {
+        return generateSignature(secret, queryString);
+    } else {
+        return generateSignature(secret, payload);
+    }
```
Comment (ldclakmal) [inline websubhub-ballerina/hub_client.bal:141]:
```suggestion
    }
    return generateSignature(secret, payload);
```

File: changelog.md [hunk 4]
```diff
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - [WebSubHub Compiler Plugin does not allow additional methods inside service declaration](https://github.com/ballerina-platform/ballerina-standard-library/issues/1417)
 - [Fix logic issue in selecting the event-type for content-publishing](https://github.com/ballerina-platform/ballerina-standard-library/issues/1460)
 - [Fix Required parameter validation for WebSubHub Service Declaration](https://github.com/ballerina-platform/ballerina-standard-library/issues/1477)
+- [Return only module specific errors from public APIs](https://github.com/ballerina-platform/ballerina-standard-library/issues/1487)
```
Comment (ThisaruGuruge) [inline changelog.md:15]: Shouldn't this go under `changed` category?

File: websubhub-ballerina/hub_listener.bal [hunk 5]
```diff
@@ -52,15 +57,15 @@ public class Listener {
...
+    # + return - An `websubhub:Error` if an error occurred during the service attaching process or else `()`
+    public isolated function attach(Service 'service, string[]|string? name = ()) returns Error? {
         if self.listenerConfig.secureSocket is () {
             log:printWarn("HTTPS is recommended but using HTTP");
         }
 
         string hubUrl = self.retrieveHubUrl(name);
         ServiceConfiguration? configuration = retrieveServiceAnnotations('service);
-        HttpToWebsubhubAdaptor adaptor = check new ('service);
+        HttpToWebsubhubAdaptor adaptor = new ('service);
```
Comment (ThisaruGuruge) [inline websubhub-ballerina/hub_listener.bal:68]:
```suggestion
        HttpToWebsubhubAdaptor adaptor = new('service);
```

File: websubhub-ballerina/hub_client.bal [hunk 6]
```diff
@@ -157,37 +133,32 @@ isolated function retrieveContentType(string? contentType, string|xml|json|byte[
...
 }
 
-# Retrieve signature for the content-distribution request payload.
-# 
-# + 'key - hashing key to be used (this is provided by the subscriber)
-# + payload - content-distribution request body
-# + return - `byte[]` containing the content signature or an `error` if there is any exception in the 
-#            function execution
-isolated function retrievePayloadSignature(string 'key, string|xml|json|byte[] payload) returns byte[]|error {
+isolated function retrievePayloadSignature(string contentType, string secret, string queryString, string|xml|json|byte[] payload) returns byte[]|error {
```
Comment (ldclakmal) [inline websubhub-ballerina/hub_client.bal:136]: Since `json` is a union of `()|boolean|int|float|decimal|string|json[]|map<json>`, we may be able to simplify `string|xml|json|byte[]` as `xml|json|byte[]`. I have observed this in other places of the module also. May be we can address this in a separate PR. WDYT?

Comment (ayeshLK) [reply websubhub-ballerina/hub_client.bal:136]: Yeah will update

**Review by ldclakmal — APPROVED** [review]

File: websubhub-ballerina/hub_client.bal (hunk 6 above)
Comment (ayeshLK) [reply websubhub-ballerina/hub_client.bal:136]: As discussed we will address this in a separate PR
